server/models/Transaction.js

```diff
@@ -411,19 +411,54 @@ export default (Sequelize, DataTypes) => {
       transaction.amountInHostCurrency = Math.round(transaction.amountInHostCurrency);
     }
 
-    const oppositeTransaction = {
-      ...transaction,
-      type: -transaction.amount > 0 ? TransactionTypes.CREDIT : TransactionTypes.DEBIT,
-      FromCollectiveId: transaction.CollectiveId,
-      CollectiveId: transaction.FromCollectiveId,
-      HostCollectiveId: await models.Collective.getHostCollectiveId(transaction.FromCollectiveId), // see https://github.com/opencollective/opencollective/issues/1154
-      amount: -transaction.netAmountInCollectiveCurrency,
-      netAmountInCollectiveCurrency: -transaction.amount,
-      amountInHostCurrency: Math.round(-transaction.netAmountInCollectiveCurrency * transaction.hostCurrencyFxRate),
-      hostFeeInHostCurrency: transaction.hostFeeInHostCurrency,
-      platformFeeInHostCurrency: transaction.platformFeeInHostCurrency,
-      paymentProcessorFeeInHostCurrency: transaction.paymentProcessorFeeInHostCurrency,
-    };
+    // Is the target "collective" (account) "Active" (has an host, manage its own budget)
+    const fromCollective = await models.Collective.findByPk(transaction.FromCollectiveId);
+
+    let oppositeTransaction;
```

Comment:
Start `oppositeTransaction` with the common fields already populated, this will simplify the following conditional outcomes.